### PR TITLE
[App Lab] Add new commands `textArea` and `slider`

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR adds the `textArea` and `slider` commands so that these design elements can be created programatically. They are already able to be created via Design mode.
This is a request by @dancodedotorg and it is not required to have this functionality surfaced to users. He requested these functions to help improve the design/flow/behavior of widgets within 'How AI Works' lessons.

The creation of the design elements are done similar to how `create` is implemented in [`slider.jsx`](https://github.com/code-dot-org/code-dot-org/blob/c94d8af9c959cf6f684fd261e91269bf2da462bf/apps/src/applab/designElements/slider.jsx#L135-L148) and [`textarea.jsx`](https://github.com/code-dot-org/code-dot-org/blob/c94d8af9c959cf6f684fd261e91269bf2da462bf/apps/src/applab/designElements/textarea.jsx#L186-L202).
These new functions are categorized as 'Advanced' in `dropletConfig.js` so that blocks do not show up for them in the toolbox.

I learned that if the function names are added to `shared_constants.rb`, then the functions are added to the Toolbox. Thus they are [NOT added](https://github.com/code-dot-org/code-dot-org/pull/53515/commits/4c7fb0f51bce3a85e43ed64e26aa91c8844fc828) to the constant `APPLAB_BLOCKS` which is the used by dashboard to initialize the default palette when creating a level, and is used by apps to determine what the full set of blocks available is.

The default values for the two functions are set in `dropletConfig.js`.
@dancodedotorg - it may be useful for other curriculum writers if documentation is provided, especially for `slider` which takes parameters `slider(id, min, max, value, step)`. Both elements can be create with default values using:
```
slider('slider1');
textArea('textArea1');
```
Or with arguments:
```
slider('slider2', 0, 10, 5, 1);
textArea('textArea2', 'example text');
```

The screencast video below demos the creation of the two Design elements:

https://github.com/code-dot-org/code-dot-org/assets/107423305/60d0d92b-fce9-4527-b8ff-c6c7c27d83e3

Shoutout to @bencodeorg for his review comments which helped me tremendously!

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-39)

## Testing story
I tested locally and confirmed that slides and text areas are now able to be added to the App screen via program code.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by dancodedotorg — APPROVED** [review]
🎉🎉🎉🎉🎉🎉🎉

File: apps/src/applab/commands.js [hunk 1]
```diff
@@ -888,6 +888,36 @@ applabCommands.textInput = function (opts) {
   return Boolean(Applab.activeScreen().appendChild(newInput));
 };
 
+applabCommands.textArea = function (opts) {
+  console.log('textArea');
```
Comment (bencodeorg) [inline apps/src/applab/commands.js]: Remove?

File: apps/src/applab/commands.js [hunk 2]
```diff
@@ -888,6 +888,36 @@ applabCommands.textInput = function (opts) {
...
+  // PARAMNAME: textArea: id vs. areaId
+  apiValidateDomIdExistence(opts, 'textArea', 'id', opts.elementId, false);
```
Comment (bencodeorg) [inline apps/src/applab/commands.js:892]: Too much work to refactor to share between textArea and textLabel?

File: apps/src/applab/commands.js [hunk 3]
```diff
@@ -947,6 +977,31 @@ applabCommands.radioButton = function (opts) {
   return Boolean(Applab.activeScreen().appendChild(newRadio));
 };
 
+applabCommands.slider = function (opts) {
+  console.log('opts', opts);
```
Comment (bencodeorg) [inline apps/src/applab/commands.js]: Remove?

File: apps/src/applab/designElements/library.js [hunk 4]
```diff
@@ -149,6 +149,8 @@ export default {
         return ElementType.IMAGE;
       case 'canvas':
         return ElementType.CANVAS;
+      case 'textarea':
```
Comment (bencodeorg) [inline apps/src/applab/designElements/library.js]: I think that "text areas" generated via the design tab are actually divs -- it looks like the ones generated by this change are real html `textarea` elements. Is that intentional?

Comment (fisher-alice) [reply apps/src/applab/designElements/library.js]: Thanks for catching this! I was basing the implementation of `textArea` function on `textLabel` within the same file. But after you pointed this out, realized that it should be implemented to how [`create`](https://github.com/code-dot-org/code-dot-org/blob/b7e03ff4a27d8bfcb696dce760bf3bb4f7e783d7/apps/src/applab/designElements/textarea.jsx#L186) is defined in `textarea.jsx`.  Updated accordingly - thank you!

File: apps/src/applab/commands.js (hunk 2 above)
Comment (fisher-alice) [reply apps/src/applab/commands.js:892]: Based on your feedback below, implementation is now different from `textLabel`.

File: apps/src/applab/commands.js [hunk 5]
```diff
@@ -888,6 +888,29 @@ applabCommands.textInput = function (opts) {
   return Boolean(Applab.activeScreen().appendChild(newInput));
 };
 
+applabCommands.textArea = function (opts) {
+  apiValidateDomIdExistence(opts, 'textArea', 'id', opts.elementId, false);
+  apiValidateType(opts, 'textArea', 'text', opts.text, 'uistring');
+  if (typeof opts.forId !== 'undefined') {
```
Comment (bencodeorg) [inline apps/src/applab/commands.js]: This looks maybe only relevant to text labels? ie, its used to associate a label with an input?

File: apps/src/applab/commands.js [hunk 6]
```diff
@@ -947,6 +970,31 @@ applabCommands.radioButton = function (opts) {
...
+  var newSlider = document.createElement('input');
+  newSlider.id = opts.elementId;
+  newSlider.setAttribute('type', 'range');
+  newSlider.min = opts.min;
+  newSlider.max = opts.max;
+  newSlider.defaultValue = opts.value;
+  newSlider.step = opts.step;
+  newSlider.style.position = 'relative';
+  newSlider.style.borderStyle = 'solid';
+  newSlider.style.width = 50;
```
Comment (bencodeorg) [inline apps/src/applab/commands.js]: It looks like the width is different than when created with design mode?

**Review by bencodeorg — COMMENTED** [review]
Side note: do you know if there needs to be an accompanying change to the docs somewhere  to cover these new functions?

**Comment by fisher-alice** [comment]
> Side note: do you know if there needs to be an accompanying change to the docs somewhere to cover these new functions?

Per @dancodedotorg 's request, he said that it is not necessary to add documentation - see [Slack thread](https://codedotorg.slack.com/archives/C03DBDN67B7/p1692891352719209?thread_ts=1692823646.226219&cid=C03DBDN67B7). This is similar to `setStyle` for which a curriculum writer can use in workspace but functionality not surfaced to students.

**Review by bencodeorg — APPROVED** [review]
LGTM!

**Pull request merged**